src/services/record.ts

```diff
@@ -174,56 +165,111 @@ export const replaceRecordService = (data: Record): Promise<readonly Selectable<
         recordKindId: recordKind.id,
         systemId: data.system_id
       },
-      false // LRU caching is not very beneficial for this operation
+      false // LRU caching is not needed for this operation
     );
 
-    // Prune existing on hold and process events for the system record
-    await Promise.all([
-      new OnHoldEventRepository(trx).prune(systemRecord.id).execute(),
-      new ProcessEventRepository(trx).prune(systemRecord.id).execute()
-    ]);
+    // Handle on hold events
+    const oheOld = (await new OnHoldEventRepository(trx).findBy({ systemRecordId: systemRecord.id }).execute()).map(
+      (ohe) => ({
+        id: ohe.id,
+        codingId: ohe.codingId,
+        ...dateTimePartsToEvent({
+          startDate: ohe.startDate,
+          startTime: ohe.startTime ?? undefined,
+          endDate: ohe.endDate ?? undefined,
+          endTime: ohe.endTime ?? undefined
+        })
+      })
+    );
+
+    const oheMatched = new Set<number>();
+    const oheAdd = (
+      await Promise.all(
+        data.on_hold_event_set.map(async (ce) => {
+          const { id: codingId } = await cacheableUpsert(new CodingRepository(trx), {
+            code: ce.coding.code,
+            codeSystem: ce.coding.code_system,
+            versionId: data.version
+          });
+          const ceNew = { codingId, ...ce.event };
+
+          // @ts-expect-error ts(7053)
+          const matched = oheOld.find((ceOld) => Object.keys(ceNew).every((key) => ceNew[key] == ceOld[key]));
```

Comment:
`ceNew[key] == ceOld[key]` should be `ceNew[key] === ceOld[key]`.

Shouldn't our linter catch `==`?